<?xml version="1.0"?>
<launch>
  <!-- Alternative launch file using depth images instead of point clouds -->

  <!-- Depth Processing Parameters -->
  <arg name="depth_scale" default="0.001" />
  <arg name="max_depth" default="10.0" />
  <arg name="min_depth" default="0.1" />
  <arg name="depth_downsample" default="2" />

  <!-- Plane Detection Parameters -->
  <arg name="min_plane_points" default="500" />
  <arg name="ransac_threshold" default="0.02" />
  <arg name="max_planes" default="5" />
  <arg name="min_plane_area" default="0.1" />
  <arg name="downsample_factor" default="4" />
  <arg name="use_rviz" default="true" />

  <!-- Depth Processor Node -->
  <node name="depth_processor_node" pkg="hsr_plane_detection" type="depth_processor_node.py" output="screen">
    <param name="depth_scale" value="$(arg depth_scale)" />
    <param name="max_depth" value="$(arg max_depth)" />
    <param name="min_depth" value="$(arg min_depth)" />
    <param name="downsample" value="$(arg depth_downsample)" />
  </node>

  <!-- Plane Detection Node (remapped to use depth-generated point cloud) -->
  <node name="plane_detection_node" pkg="hsr_plane_detection" type="plane_detection_node.py" output="screen">
    <remap from="/hsrb/head_rgbd_sensor/points" to="/depth_point_cloud" />
    <param name="min_plane_points" value="$(arg min_plane_points)" />
    <param name="ransac_threshold" value="$(arg ransac_threshold)" />
    <param name="max_planes" value="$(arg max_planes)" />
    <param name="min_plane_area" value="$(arg min_plane_area)" />
    <param name="downsample_factor" value="$(arg downsample_factor)" />
  </node>

  <!-- RViz for Visualization -->
  <node if="$(arg use_rviz)" name="rviz" pkg="rviz" type="rviz"
        args="-d $(find hsr_plane_detection)/rviz/plane_detection.rviz"
        output="screen" />

</launch>
